Src/BootCamp.Chapter/Item.cs

```diff
@@ -1,26 +1,25 @@
-﻿namespace BootCamp.Chapter
+﻿using System;
+
+namespace BootCamp.Chapter
 {
     public class Item
     {
-        private string _name;
-        public string GetName()
-        {
-            return _name;
-        }
-
-        private decimal _price;
-        public decimal GetPrice()
-        {
-            return _price;
-        }
+        public string Name { get; }
 
-        private float _weight;
+        public decimal Price { get; }
 
+        public float Weight { get; }
         public Item(string name, decimal price, float weight)
         {
-            _name = name;
-            _price = price;
-            _weight = weight;
+            Name = name ?? throw new ArgumentNullException("name");
+            Price = price;
+            Weight = weight;
+        }
+        public bool Equals(Item b)
+        {
+            if (this.Name == (b?.Name??throw new ArgumentNullException()) && this.Price ==( b?.Price??0) && this.Weight ==(b?.Weight??0))
```

Comment:
Very hard to read line. The argument null exception is better thrown in item constructor, to not let a null item in the first place. I would just compare name and price and weight and be done with it.

Reply:
You don't have to recheck for item being null. Just do it once. Also one null item can be equal to another null item- that's fine.